tests/test-data-store.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * Tests for the WC_Order_Data_Store_Custom_Table class.
+ *
+ * @package Woocommerce_Order_Tables
+ * @author  Liquid Web
+ */
+
+class DataStoreTest extends TestCase {
+
+	/**
+	 * Fire the necessary actions to bootstrap WordPress.
+	 *
+	 * @before
+	 */
+	public function init() {
+		do_action( 'init' );
+	}
+
+	/**
+	 * Remove any closures that have been assigned to filters.
+	 *
+	 * @after
+	 */
+	public function remove_filter_callbacks() {
+		remove_all_filters( 'woocommerce_shop_order_search_fields' );
+	}
+
+	public function test_create() {
+		$instance  = new WC_Order_Data_Store_Custom_Table();
+		$order     = $this->factory()->order->create_and_get();
+		$order_key = 'my_custom_order_key';
+
+		add_filter( 'woocommerce_generate_order_key', function () use ( $order_key ) {
+			return $order_key;
+		} );
+
+		$instance->create( $order );
+
+		$this->assertEquals( 'wc_' . $order_key, $order->get_order_key() );
+	}
+
+	public function test_get_order_count() {
+		$orders = $this->factory()->order->create_many( 5, array(
+			'post_status' => 'wc-pending',
+		) );
+
+		$this->assertEquals(
+			count( $orders ),
+			( new WC_Order_Data_Store_Custom_Table() )->get_order_count( 'wc-pending' )
```

Comment:
This does not appear to be valid syntax for PHP 5.3 based on the Travis job.

Reply:
Confirmed - definitely an issue with how 5.3 allows instantiation and using an operator immediately - https://stackoverflow.com/a/21743040/1400048